c2rust-analyze/src/pointer_id.rs

```diff
@@ -60,21 +59,55 @@ impl fmt::Debug for PointerId {
     }
 }
 
+impl PartialEq for PointerId {
+    fn eq(&self, other: &PointerId) -> bool {
+        self.index() == other.index()
+    }
+
+    fn ne(&self, other: &PointerId) -> bool {
+        self.index() != other.index()
+    }
+}
+
+impl Eq for PointerId {}
+
+impl PartialOrd for PointerId {
+    fn partial_cmp(&self, other: &PointerId) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for PointerId {
+    fn cmp(&self, other: &PointerId) -> Ordering {
+        self.index().cmp(&other.index())
+    }
+}
+
+impl Hash for PointerId {
+    fn hash<H: Hasher>(&self, hasher: &mut H) {
+        self.index().hash(hasher);
+    }
+}
+
 #[derive(Clone, Debug)]
 pub struct NextLocalPointerId(u32);
 
 impl NextLocalPointerId {
-    pub fn new() -> NextLocalPointerId {
+    pub fn _new() -> NextLocalPointerId {
```

Comment:
Why the rename?

Reply:
The function is currently unused, but I've left it in place for completeness of the API (`NextGlobalPointerId` has a `fn new()`, so the local version should too) and in case it's needed again in the future.  To suppress the dead code warning, I prefer a leading underscore rather than `#[allow(dead_code)]` because it makes it more likely that anyone who starts using the function in the future will also remember to un-suppress the warning.